src/IdempotentAPI.AccessCache/Lockers/InProcessAccessLock.cs

```diff
@@ -15,44 +17,53 @@ namespace IdempotentAPI.AccessCache.Lockers
     /// </example>
     internal sealed class InProcessAccessLock : IDisposable
     {
-        private static readonly object localLocker = new object();
-        private static readonly Dictionary<string, LockDisposeTracker> lockValueMapper = new Dictionary<string, LockDisposeTracker>();
+        private static readonly ConcurrentDictionary<string, LockDisposeTracker> lockValueMapper = new ();
 
         private readonly string _lockedValue;
 
         public InProcessAccessLock(string valueToLock)
         {
             _lockedValue = valueToLock;
+        }
 
-            LockDisposeTracker disposeTracker;
-            lock (localLocker)
+        public static async Task<InProcessAccessLock> CreateAsync(string valueToLock)
+        {
+            var disposeTracker = lockValueMapper.AddOrUpdate(valueToLock, (_) => new LockDisposeTracker(), (_, disposeTracker) => disposeTracker with
             {
-                if (!lockValueMapper.TryGetValue(valueToLock, out disposeTracker))
-                {
-                    disposeTracker = new LockDisposeTracker();
-                    lockValueMapper.Add(valueToLock, disposeTracker);
-                }
-                disposeTracker.Count++;
-            }
-            Monitor.Enter(disposeTracker);
+                Count = disposeTracker.Count + 1
+            });
+            await disposeTracker.Semaphore.Value.WaitAsync();
+            return new InProcessAccessLock(valueToLock);
         }
 
         public void Dispose()
         {
-            lock (localLocker)
+            while (true)
```

Comment:
Why is there a need for a `while (true)` here?

Reply:
@ikyriak  To ensure that value was updated in concurrent dictionary
basically the flow is following 
loop until it get updated
`TryUpdate` accepts original value and updated value and return false if original value was changed from other thread

Reply:
@ikyriak Hi, do you have any other questions ?

Reply:
@dimmy-timmy Thank you for your response. I will include your improvements in the `2.1.0-rc-01` release.